### PR TITLE
Add SDK methods for org invites

### PR DESCRIPTION
**Pull request by iamsolankiamit**
See: https://www.flowdock.com/app/rulemotion/p-organizations/threads/o043xZcnvf1tTjmjykyulfIYP14
Connects-to: #1007
Depends-on: https://github.com/balena-io/balena-api/issues/2712
Depends-on: https://github.com/balena-io/balena-api/pull/2750
Depends-on: https://github.com/balena-io-modules/balena-errors/pull/46
Change-type: minor
Signed-off-by: Amit Solanki <amit@balena.io>

Resolves: # 
HQ: <url> 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/models/organization-invite.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+		 * @description
+		 * This method returns all invites for a specific organization.
+		 *
+		 * @param {String|Number} handleOrId - organization handle (string), or id (number)
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object[]} - invites
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.organization.invite.getAllByOrganization('MyApp').then(function(invites) {
```
Comment (thgreasi) [inline lib/models/organization-invite.ts:88]:
```suggestion
		 * balena.models.organization.invite.getAllByOrganization('myorganization').then(function(invites) {
```

File: lib/models/organization-invite.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+		 * @param {Object} options - invite creation parameters
+		 * @param {String} options.invitee - the email/balena_username of the invitee
+		 * @param {String} [options.roleName="developer"] - the role name to be granted to the invitee
+		 * @param {String} [message=null] - the message to send along with the invite
+		 *
+		 * @fulfil {String} - organization invite
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.organization.invite.create('MyApp', { invitee: "invitee@example.org", roleName: "developer", message: "join my app" }).then(function(invite) {
```
Comment (thgreasi) [inline lib/models/organization-invite.ts:136]:
```suggestion
		 * balena.models.organization.invite.create('myorganization', { invitee: "invitee@example.org", roleName: "developer", message: "join my org" }).then(function(invite) {
```

File: tests/integration/models/organizationInvite.spec.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
+// tslint:disable-next-line:import-blacklist
+import * as m from 'mochainon';
+import * as parallel from 'mocha.parallel';
+import { balena, givenInitialOrganization, givenLoggedInUser } from '../setup';
+import { timeSuite } from '../../util';
+import type * as BalenaSdk from '../../../typings/balena-sdk';
+const { expect } = m.chai;
+import { assertDeepMatchAndLength } from '../../util';
+const TEST_EMAIL = 'user.test@example.org';
+const TEST_MESSAGE = 'Hey!, Join my app on balenaCloud';
```
Comment (thgreasi) [inline tests/integration/models/organizationInvite.spec.ts:10]:
```suggestion
const TEST_MESSAGE = 'Hey!, Join my org on balenaCloud';
```

File: tests/integration/models/organizationInvite.spec.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+						{
+							invitee: TEST_EMAIL,
+							roleName: TEST_ROLE,
+							message: TEST_MESSAGE,
+						},
+					);
+				});
+
+				parallel(
+					'balena.models.organization.invite.getAllOrganization()',
```
Comment (thgreasi) [inline tests/integration/models/organizationInvite.spec.ts:143]:
```suggestion
					'balena.models.organization.invite.getAllByOrganization()',
```

**Review by thgreasi — COMMENTED** [review]
Just NITs, other than that LGTM :+1:

**Review by thgreasi — APPROVED** [review]
LGTM to get merged as soon as the API is deployed and the tests pass :+1:

**Comment by thgreasi** [comment]
@balena-ci rebase

**Comment by iamsolankiamit** [comment]
@balena-ci rebase

**Pull request merged**